rousseau_vote/lib/src/widgets/user/badge_slider.dart

```diff
@@ -65,4 +67,53 @@ class BadgeSlider extends StatelessWidget {
       ),
     );
   }
+
+  void showMeritMeaning(String merit, BuildContext context){
+    showDialog<CupertinoAlertDialog>(
+      context: context,
+      builder: (BuildContext context) {
+        return CupertinoAlertDialog(
+          content: Column(
+            children: <Widget>[
+              Image( 
+                image: AssetImage('assets/merits/' + merit +  '_true.png'),
+                width: 60,
+                height: 60,
+              ),
+              Padding(
+                padding: const EdgeInsets.only(top:15),
+                child: Text(RousseauLocalizations.getText(context, merit)),
+              )
+            ],
+          ),
+        );
+      }
+    );
+  }
+
+  List<String> badgesToMerits(){
+    List<String> meritIcons = [];
+    if (badges.isNotEmpty){
+      for(final badge in badges){
+        switch(badge.code) {
+          case 'list_representative':
+          case 'italia_cinque_stelle_volunteer':
+          case 'villaggio_rousseau_volunteer': if(!meritIcons.contains('merit1')) meritIcons.add('merit1'); break;
+          case 'call_to_actions_organizer':
+          case 'activism_organizer':
+          case 'sharing_proposer':
+          case 'user_lex_proposer': if(!meritIcons.contains('merit2')) meritIcons.add('merit2'); break; 
+          case 'graduate': meritIcons.add('merit3'); break;
+          case 'english_language_expert': meritIcons.add('merit4'); break;
+          case 'openday_participant':
+          case 'villaggio_rousseau_participant': if(!meritIcons.contains('merit5')) meritIcons.add('merit5'); break;
+          case 'elearnign_student': meritIcons.add('merit6'); break;
+          case 'special_mentions': meritIcons.add('merit7'); break;
+          case 'high_specialization': meritIcons.add('merit8'); break;
+          case 'community_leader': meritIcons.add('merit9'); break;
```

Comment:
Prefer a declarative approach:
```
const Map<String, String> BADGE_ICON_MAPPING = { 'list_representative': 'merit_1', ... 'elearnign_student': 'merit_6};

...

for(final badge in badges) {
  final String meritIcon = BADGE_ICON_MAPPING[bagde]
  if(!meritIcons.contains(meritIcon)) {
    meritIcons.add(meritIcon);
  }
}
```

Reply:
ok, I have switched to your suggested declarative approach. What if a badge.code is not present in the BADGE_ICON_MAPPING keys?